skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,568 @@
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.intervals = [(1, 3), (4, 7)]
+        self.assertEqual(f.intervals, [(1, 3), (4, 7)])
+        self.assertEqual(im._is_stale_tree, True)
+
+    def test_set_interval_none(self):
+        with self.assertRaises(RuntimeError):
+            f = Interval(interval_metadata=None,
+                         intervals=[(1, 2), (4, 7)],
+                         boundaries=[(True, False), (False, False)],
+                         metadata={'name': 'sagA', 'function': 'transport'})
+            f.intervals = [(1, 3), (4, 7)]
+
+    def test_set_metadata(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.metadata = {'name': 'sagB', 'function': 'transport'}
+        self.assertDictEqual(f.metadata,
+                             {'name': 'sagB', 'function': 'transport'})
+        feats = list(im.query([(1, 2)]))
+        self.assertDictEqual(feats[0].metadata,
+                             {'name': 'sagB', 'function': 'transport'})
+
+
+class TestIntervalUtil(unittest.TestCase):
+    def test_assert_valid_interval_tuple(self):
+        interval = (1, 2)
+        _assert_valid_interval(interval)
+        st, end = interval
+        self.assertEqual(st, 1)
+        self.assertEqual(end, 2)
+
+    def test_assert_valid_interval_tuple_swapped(self):
+        interval = (2, 1)
+        with self.assertRaises(ValueError):
+            _assert_valid_interval(interval)
+
+    def test_assert_valid_interval_tuple_bad(self):
+        with self.assertRaises(ValueError):
+            _assert_valid_interval((1, 2, 3))
+
+    def test_assert_valid_boundary_tuple(self):
+        boundary = (True, False)
+        _assert_valid_boundary(boundary)
+        st, end = boundary
+        self.assertEqual(st, True)
+        self.assertEqual(end, False)
+
+    def test_assert_valid_boundary_tuple_bad(self):
+        boundary = (True, False, True)
+        with self.assertRaises(ValueError):
+            _assert_valid_boundary(boundary)
+
+
+class TestIntervalMetadata(unittest.TestCase):
+
+    def test_init(self):
+        im = IntervalMetadata()
+        self.assertEqual(im._is_stale_tree, False)
+
+    def test_add(self):
+        im = IntervalMetadata()
+        im.add(intervals=[(1, 2), (4, 7)],
+               metadata={'gene': 'sagA',  'location': 0})
+
+        self.assertEqual(im._metadata[0].intervals,
+                         [(1, 2), (4, 7)])
+        self.assertEqual(im._metadata[0].metadata,
+                         {'gene': 'sagA', 'location': 0})
+        self.assertTrue(im._intervals is not None)
```

Comment:
I'm assuming that the state represented by `im._intervals` is deterministic. Would it be possible to change `self.assertTrue(im_intervals is not None)` to test the expected state of `im._intervals`?